src/dappsExplorer/DAppsExplorerScreenSearchFilter.tsx

```diff
@@ -79,6 +82,27 @@ export function DAppsExplorerScreenSearchFilter() {
   const [favoriteResultsEmpty, setFavoriteResultsEmpty] = useState(false)
   const [allResultEmpty, setAllResultEmpty] = useState(false)
 
+  // Hide favorites when there are no favorites or no favorites matching the search term
+  useEffect(() => {
```

Comment:
could we simplify this logic?
```
const filteredFavorites = favoriteDappsWithCategoryNames.filter((dapp) =>
              dapp.categories.includes(selectedFilter)
            )
const searchResults = searchDappList(filteredFavorites, searchTerm)
setFavoriteResultsEmpty(searchDappList(filteredFavorites, searchTerm).length === 0)
```
slightly less performant but since favoriteDappsWithCategoryNames is not a very large list, the readability seems worth it?

if you're worried about performance, `favoriteResultsEmpty` could be a `useMemo` instead of a `useEffect`?

Reply:
Trying the code above I am getting the following:
`Error: Too many re-renders. React limits the number of renders to prevent an infinite loop.` 😕